packages/@yoda/httpdns/src/httpdns.cc

```diff
@@ -0,0 +1,164 @@
+#include "httpdns.h"
+#include <node_api.h>
+#include <uv.h>
+#include <stdio.h>
+#include <common.h>
+#include <string.h>
+#include <mutex>
+#include <list>
+
+using namespace std;
+static bool g_httpdns_initd = false;
+static uv_async_t async;
+
+class HttpdnsAsyncTask {
+ public:
+  napi_env env;
+  napi_ref callback;
+  uv_async_t async_handle;
+  int status;
+
+ public:
+  ~HttpdnsAsyncTask() {
+    if (env && callback) {
+      NAPI_CALL_RETURN_VOID(env, napi_delete_reference(env, callback));
+    }
+  }
+};
+
+static int notifyFinishedNotify(int status, void* userdata) {
+  HttpdnsAsyncTask* task = (HttpdnsAsyncTask*)userdata;
+  task->status = status;
+  uv_async_send(&task->async_handle);
+}
+
+static void handleFinishedService(uv_async_t* handle) {
+  HttpdnsAsyncTask* task = (HttpdnsAsyncTask*)handle->data;
+  napi_env env = task->env;
+  napi_ref reference = task->callback;
+  napi_value fun;
+  napi_handle_scope scope;
+  napi_value global;
+  napi_value argv[1];
+
+  napi_open_handle_scope(env, &scope);
+  napi_get_reference_value(env, reference, &fun);
+  napi_get_global(env, &global);
+  napi_get_boolean(env, 0 == task->status, &argv[0]);
```

Comment:
Error should be an `Error` with a message, see https://nodejs.org/dist/latest-v10.x/docs/api/n-api.html#n_api_napi_create_error.

Reply:
```c
napi_vlue argv[2];

if (0 != task->status) {
  napi_value errorMsg;
  napi_create_string_utf8(env, "sync failed", AUTO_LENGTH, &errorMsg);
  napi_create_error(env, nullptr, errorMsg, &argv[0]);
  napi_get_boolean(env, false, &argv[1]);
} else {
  napi_get_undefined(env, &argv[0]);
  napi_get_boolean(env, true, &argv[1]);
}
```